aten/src/ATen/cuda/CUDAGraph.cu

```diff
@@ -0,0 +1,38 @@
+#include <ATen/cuda/CUDAGraph.h>
+
+namespace at::cuda {
+
+__global__ void dynamic_graph_updater_kernel(cudaGraphKernelNodeUpdate* updates, size_t num_updates) {
+    constexpr size_t buf_sz = 1024;
+    __shared__ const void* pointer_indirection[buf_sz];
+    for (size_t off = 0; off < num_updates; off += buf_sz) {
```

Comment:
For sure - but - in this approach it'll never have two different threads trying to update the same kernel node at the same time. In theory, it shouldn't race, because it's always modifying different params of the node. Do you know if that's safe?
Also do you know if there's a practical maximum number of updates here? I picked 1024 completely at random.